Orchestrion GA (#26329)
DataDog/documentation: content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/go.md

   go install github.com/DataDog/orchestrion@latest
   ```

>    <div class="alert alert-info">Ensure <code>$(go env GOBIN)</code> (or <code>$(go env GOPATH)/bin</code>) is in your <code>$PATH</code>.</div>

Reviewer: We had discussed this in the previous iteration... The issue is that how to actually do this can vary a lot depending on platform + other user settings... It ended up looking messy, and this really is just here to make sure the `orchestrion` command is indeed visible in the shell...

Also -- if you're going to be doing this in your CI/CD pipeline, you probably want to make the `$PATH` change at the same time as this is done (if necessary)... So having it here avoids folks forget about that step.

>    <div class="alert alert-info">Ensure <code>$(go env GOBIN)</code> (or <code>$(go env GOPATH)/bin</code>) is in your <code>$PATH</code>.</div>

Reviewer 2: Gotcha. Thanks for the clarification!

1. Register Orchestrion in your project's `go.mod`:
   ```sh
   orchestrion pin
   ```
   Refer to the output of `orchestrion pin -help` for more information about available customization options.
1. Commit changes to your version control system (unless you are integrating `orchestrion` directly in your CI/CD pipeline):
   ```sh
   git commit -m "chore: enable orchestrion" go.mod go.sum orchestrion.tool.go
   ```

>    Now you can manage your dependency on `orchestrion` like any other dependency using the `go.mod` file.

Reviewer: I indented it into the list item because this is only accurate/relevant if you've checked in the changes (meaning you're not doing a "pure CI/CD integration").

That being said, the distinction is subtle and I wouldn't mind un-indenting that if you still think it's better.

>    Now you can manage your dependency on `orchestrion` like any other dependency using the `go.mod` file.

Reviewer 2: Ahh, I see what you mean... I think it's fine as-is then. :)

#### Use the continuous profiler

Your Orchestrion-built application includes [continuous profiler][12] instrumentation.
To enable the profiler, set the environment variable `DD_PROFILING_ENABLED=true` at runtime.

### Troubleshooting

To troubleshoot `orchestrion`-managed builds, see [Troubleshooting Go Compile-Time Instrumentation][13]

[4]: https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/ddtrace
[6]: https://github.com/DataDog/orchestrion
[7]: /security/application_security/threats/exploit-prevention
[8]: https://go.dev/doc/devel/release#policy
[10]: https://pkg.go.dev/cmd/go#hdr-Modules__module_versions__and_more
[11]: https://github.com/DataDog/orchestrion/releases
[12]: /profiler
[13]: /tracing/troubleshooting/go_compile_time/
[14]: /getting_started/tagging/unified_service_tagging/
[15]: /security/application_security/threats/exploit-prevention/

{{% /tab %}}

{{% tab "Manual instrumentation" %}}

### Activate Go integrations to create spans

> Datadog has a series of pluggable packages which provide out-of-the-box support for instrumenting a series of libraries and frameworks. A list of these packages can be found in the [Compatibility Requirements][1] page. Import these packages into your application and follow the configuration instructions listed alongside each [Integration][1].

Reviewer 2: ```suggestion
Datadog has a series of pluggable packages which provide out-of-the-box support for instrumenting a series of libraries and frameworks. You can find a list of these packages in the [Compatibility Requirements][1] page. Import these packages into your application and follow the configuration instructions listed alongside each [Integration][1].
```

> Datadog has a series of pluggable packages which provide out-of-the-box support for instrumenting a series of libraries and frameworks. A list of these packages can be found in the [Compatibility Requirements][1] page. Import these packages into your application and follow the configuration instructions listed alongside each [Integration][1].

Reviewer 2: Is `[Integration][1]` a mistake here?

Reviewer: I guess the "pluggable packages" are also called "integrations" (or "contribs", although that's not a name I'd use in the docs), and they are still listed at `[1]`.

Reviewer 2: I don't think it makes sense to link to the same exact link twice in the same paragraph. Also, assuming this "integration" is different from a Datadog integration (which it seems to be), I would like to use a different word. If it's "pluggable packages", so be it.

Reviewer: Changing to `package` (avoiding to repeat "pluggable" because what does it even mean?)

> Datadog has a series of pluggable packages which provide out-of-the-box support for instrumenting a series of libraries and frameworks. A list of these packages can be found in the [Compatibility Requirements][1] page. Import these packages into your application and follow the configuration instructions listed alongside each [Integration][1].

Reviewer: That verbiage is part of the historical manual instrumentation steps that was just moved round (from being 1st to being 2nd). I must admit I also don't love that phrasing but it's not in my scope so I don't feel legitimate to make recommendations here.

@vidurkhannadatadog might be able to provide some ideas?

### Configuration

If needed, configure the tracing library to send application performance telemetry data as you require, including setting up Unified Service Tagging. Read [Library Configuration][3] for details.

> For configuration instructions and details about using the API, see the Datadog [API documentation][4].

Reviewer 2: ```suggestion
For configuration instructions and details about using the API, see the [Datadog API documentation][4].
```

[1]: /tracing/compatibility_requirements/go
[3]: /tracing/trace_collection/library_config/go/
[4]: https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/ddtrace

> [5]: /tracing/trace_collection/automatic_instrumentation/?tab=datadoglibraries#install-and-configure-the-agent

Reviewer 2: Is this link being used?

}
{{</code-block>}}

#### Prevent instrumentation of some code

> The `//orchestrion:ignore` directive can be used to prevent `orchestrion` from performing _any_ modification on the annotated code.

Reviewer 2: ```suggestion
You can use the `//orchestrion:ignore` directive to prevent `orchestrion` from performing _any_ modification on the annotated code.
```
